### PR TITLE
Js runtime host integration

### PR DESCRIPTION
**Pull request by CedricGuillemet**

### CODE REVIEW

File: .github/jobs/linux.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
     - script: |
         mkdir build
         cd build
-        cmake .. -GNinja -DJSCORE_LIBRARY=/usr/lib/x86_64-linux-gnu/libjavascriptcoregtk-4.0.so -DNAPI_JAVASCRIPT_ENGINE=${{ parameters.JSEngine }} -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBGFX_CONFIG_DEBUG=ON -DCMAKE_UNITY_BUILD=${UNITY_BUILD}
+        cmake .. -G Ninja -D JAVASCRIPTCORE_LIBRARY=/usr/lib/x86_64-linux-gnu/libjavascriptcoregtk-4.0.so -D NAPI_JAVASCRIPT_ENGINE=${{ parameters.JSEngine }} -D CMAKE_BUILD_TYPE=RelWithDebInfo -D BGFX_CONFIG_DEBUG=ON -DCMAKE_UNITY_BUILD=${UNITY_BUILD}
```
Comment (docEdub) [inline .github/jobs/linux.yml:35]:
```suggestion
        cmake .. -G Ninja -D JAVASCRIPTCORE_LIBRARY=/usr/lib/x86_64-linux-gnu/libjavascriptcoregtk-4.0.so -D NAPI_JAVASCRIPT_ENGINE=${{ parameters.JSEngine }} -D CMAKE_BUILD_TYPE=RelWithDebInfo -D BGFX_CONFIG_DEBUG=ON -D CMAKE_UNITY_BUILD=${UNITY_BUILD}
```

**Review by docEdub — COMMENTED** [review]
👍 LGTM

File: .github/jobs/ios.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ jobs:
     - script: |
         mkdir buildiOS
         cd buildiOS
-        cmake .. -G Xcode -DCMAKE_TOOLCHAIN_FILE=../Dependencies/ios-cmake/ios.toolchain.cmake -DPLATFORM=OS64COMBINED -DENABLE_ARC=0 -DDEPLOYMENT_TARGET=${{ parameters.deploymentTarget }} -DENABLE_GLSLANG_BINARIES=OFF -DSPIRV_CROSS_CLI=OFF -DCMAKE_UNITY_BUILD=${UNITY_BUILD}
+        cmake .. -G Xcode -D IOS=ON -D DEPLOYMENT_TARGET=${{ parameters.deploymentTarget }} -D ENABLE_GLSLANG_BINARIES=OFF -D SPIRV_CROSS_CLI=OFF -D CMAKE_UNITY_BUILD=${UNITY_BUILD}
```
Comment (bghgary) [inline .github/jobs/ios.yml:30]: > -D ENABLE_GLSLANG_BINARIES=OFF -D SPIRV_CROSS_CLI=OFF

Is this still necessary?

File: Apps/Playground/CMakeLists.txt [hunk 3]
```diff
@@ -134,6 +133,11 @@ if(WIN32)
         PRIVATE "shlwapi.lib")
 endif()
 
+# See https://gitlab.kitware.com/cmake/cmake/-/issues/23543
+# If we can set minimum required to 3.26+, then we can use the `copy -t` syntax instead.
+add_custom_command(TARGET Playground POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E $<IF:$<BOOL:$<TARGET_RUNTIME_DLLS:Playground>>,copy,true> $<TARGET_RUNTIME_DLLS:Playground> $<TARGET_FILE_DIR:Playground>)
```
Comment (bghgary) [inline Apps/Playground/CMakeLists.txt:139]: This code replaces the need for the CMake extension function `target_link_to_dependencies` and associated.

File: CMakeLists.txt [hunk 4]
```diff
@@ -6,10 +6,16 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 include(FetchContent)
 FetchContent_Declare(cmake-extensions
         GIT_REPOSITORY https://github.com/BabylonJS/CMakeExtensions.git
```
Comment (bghgary) [inline CMakeLists.txt:8]: What's up with the double tabbing?

File: Core/AppRuntime/CMakeLists.txt
Comment (bghgary) [inline Core/AppRuntime/CMakeLists.txt:1]: Make sure the changes that were made here after copying to JsRuntimeHost are reflected in JsRuntimeHost. Ditto for other submodules.

**Review by bghgary — APPROVED** [review]

File: CMakeLists.txt (hunk 4 above)
Comment (CedricGuillemet) [reply CMakeLists.txt]: Fixed.

File: .github/jobs/ios.yml (hunk 2 above)
Comment (CedricGuillemet) [reply .github/jobs/ios.yml]: Nop. Removed.

File: Core/AppRuntime/CMakeLists.txt
Comment (CedricGuillemet) [reply Core/AppRuntime/CMakeLists.txt:1]: SHA updated to the latest

File: Apps/Playground/CMakeLists.txt (hunk 3 above)
Comment (CedricGuillemet) [reply Apps/Playground/CMakeLists.txt:134]: call to `target_link_to_dependencies` removed.

Comment (bghgary) [reply Apps/Playground/CMakeLists.txt:134]: We should also remove the related functions in CMakeExtensions repo itself.

**Pull request merged**